__tests__/Unit/Components/Tasks/TaskDetails.test.tsx

```diff
@@ -659,3 +643,78 @@ describe('TaskDependency', () => {
+
+jest.mock('next/router', () => ({
+    useRouter() {
+        return {
+            query: {
+                dev: 'true',
+            },
+            push: mockNavigateToUpdateProgressPage,
+        };
+    },
+}));
+
+describe('Details component', () => {
+    it('Renders the detail type and value correctly', () => {
+        renderWithRouter(<Details detailType="Assignee" value="John Doe" />);
+        const detailType = screen.getByText('Assignee:');
+        const detailValue = screen.getByText('John Doe');
+        expect(detailType).toBeInTheDocument();
+        expect(detailValue).toBeInTheDocument();
+    });
+
+    it('Renders a GitHub link correctly', () => {
+        const githubIssueUrl = 'https://github.com/example/repo/issues/123';
+        renderWithRouter(<Details detailType="Link" value={githubIssueUrl} />);
+        const link = screen.getByRole('link', { name: /Open GitHub Issue/i });
+        expect(link).toBeInTheDocument();
+        expect(link).toHaveAttribute('href', githubIssueUrl);
+    });
+
+    it('Does not display tooltip on hover when timestamp is not provided', async () => {
+        render(<Details detailType="Started On" value="N/A" />);
+        const detailValue = screen.getByText('Started:');
+        fireEvent.mouseOver(detailValue);
+        const tooltip = screen.queryByText('N/A', { selector: '.tooltip' });
+        expect(tooltip).not.toBeInTheDocument();
+    });
+
+    it('Renders an extension request icon for Ends On with URL', () => {
+        const url = 'https://example.com';
+        renderWithRouter(<Details detailType="Ends On" url={url} />);
+        const extensionRequestIcon = screen.getByTestId(
+            'extension-request-icon'
+        );
+        expect(extensionRequestIcon).toBeInTheDocument();
+        expect(extensionRequestIcon).toHaveAttribute('href', url);
+    });
+
+    it('Does not render an extension request icon for Ends On without URL', () => {
+        renderWithRouter(<Details detailType="Ends On" />);
+        const extensionRequestIcon = screen.queryByTestId(
+            'extension-request-icon'
+        );
+        expect(extensionRequestIcon).not.toBeInTheDocument();
+    });
+    it('Renders a tooltip for timestamps', async () => {
+        const task = {
+            id: 'L1SDW6O835o0EI8ZmvRc',
+            startedOn: '1707869428.523',
+        };
+        renderWithRouter(
+            <Details detailType="Started On" value={task.startedOn} />
+        );
+        try {
+            const detailTypeElement = screen.getByText('Started:');
+            expect(detailTypeElement).toBeInTheDocument();
+            fireEvent.mouseOver(detailTypeElement);
+            const tooltip = await screen.findByTestId('tooltip');
+            expect(tooltip).toBeInTheDocument();
+            expect(tooltip.textContent).toBeTruthy();
+        } catch (error) {
+            console.error('Error occurred during tooltip rendering:', error);
+            throw error;
+        }
```

Comment:
What's the need of try catch block in tests, not getting🤔?

Reply:
The `try-catch` block in the test code is included to handle any unexpected errors that might occur during the execution of the test. It ensures that if an error occurs during the rendering of the tooltip, it will be caught and logged to the console for debugging purposes. While it might not always be necessary, it can help identify and troubleshoot issues, especially in asynchronous code like this where we're awaiting the rendering of the tooltip. If you believe it's not needed or if there's a better approach, I'm open to suggestions on how to improve the test code.

Reply:
@yesyash said we can use this `try-catch` in the tests